app/src/main/java/org/onionshare/android/files/FileManager.kt

```diff
@@ -42,27 +41,33 @@ class FileManager @Inject constructor(
             val sizeHuman = if (size == 0L) ctx.getString(R.string.unknown) else {
                 formatShortFileSize(ctx, size)
             }
-            SendFile(name, sizeHuman, uri)
+            SendFile(name, sizeHuman, size, uri, documentFile.type)
         }
         return FilesAdded(existingFiles + files)
     }
 
-    fun zipFiles(state: FilesAdded): FilesReadyForDownload {
+    fun zipFiles(files: List<SendFile>, ensureActive: () -> Unit): FilesReadyForDownload {
```

Comment:
That is actually how it was before. I had changed it, because I found it a bit strange to hand over coroutine contexts in a function, but maybe that's fine.

What might be better and seems to be possible is using `currentCoroutineContext().ensureActive()` which can be done when we make this a suspend function.